### PR TITLE
Adding additional node in the node-hub

### PR DESCRIPTION
**Pull request by haixuanTao**
In order to simplify testing with a terminal, I have added couple of simple nodes in the node-hub.

To use:
```bash
cd examples/terminal-dataflow
dora build dataflow.yml
dora start dataflow.yml

# Terminal A
terminal-print

# Terminal B
pyarrow-sender


```

Note that there might be issues with stopping the dataflow and you might need to use `dora destroy`

# Video example

[Screencast from 2024-08-09 12-41-17.webm](https://github.com/user-attachments/assets/04db8e4f-652e-42a9-a42e-6c49d2ff9fae)

### CODE REVIEW

**Comment by heyong4725** [comment]
@haixuanTao Can you make dynamic node example that the node can handle both input and output. Basically it is similar to LLM ReAct pattern: User <---> LLM, the User take keyboard input, send to LLM as query prompts, and LLM send back the inference result back to the User and print out the result

**Comment by haixuanTao** [comment]
Sure thing!

**Review by phil-opp — APPROVED** [review]

**Comment by haixuanTao** [comment]
[Screencast from 2024-08-13 11-41-16.webm](https://github.com/user-attachments/assets/aa77727a-ff3b-4362-9b4f-78c4d5c5aa02)

So I have added:
- a new node called `terminal-input` that can both send and receive data.
- a node called `dora-echo` that is just echoing the data received and should showcase how to both send and receive data from the same node.

I'm having some technical issues with using LLM due to VPN issues and i'm afraid it might be not reproducible but the `dora-echo` node should be enough to generalize on LLM nodes.

File: node-hub/dora-microphone/dora_microphone_save/recorded_audio.wav
Comment (Hennzau) [inline node-hub/dora-microphone/dora_microphone_save/recorded_audio.wav:1]: Hi! Looks good to me. Be careful, you left some audio files in the commit.

**Comment by heyong4725** [comment]
I tried this terminal example, it can echo back the digit, but for string, it give some errors. I guess it is related to Python:

'''
(py311) demo@YHE-RM05:~/src/dora-rs/test/dora/examples/echo$ terminal-input
Provide the data you want to send:  1
Received: [1]
Provide the data you want to send:  2
Received: [2]
Provide the data you want to send:  3
Received: [3]
Provide the data you want to send:  hello
Traceback (most recent call last):
  File "/home/demo/anaconda3/envs/py311/bin/terminal-input", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/demo/src/dora-rs/test/dora/node-hub/terminal-input/terminal_input/main.py", line 45, in main
    data = ast.literal_eval(data)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/demo/anaconda3/envs/py311/lib/python3.11/ast.py", line 110, in literal_eval
    return _convert(node_or_string)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/demo/anaconda3/envs/py311/lib/python3.11/ast.py", line 109, in _convert
    return _convert_signed_num(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/demo/anaconda3/envs/py311/lib/python3.11/ast.py", line 83, in _convert_signed_num
    return _convert_num(node)
           ^^^^^^^^^^^^^^^^^^
  File "/home/demo/anaconda3/envs/py311/lib/python3.11/ast.py", line 74, in _convert_num
    _raise_malformed_node(node)
  File "/home/demo/anaconda3/envs/py311/lib/python3.11/ast.py", line 71, in _raise_malformed_node
    raise ValueError(msg + f': {node!r}')
ValueError: malformed node or string on line 1: <ast.Name object at 0x7f643daa96c0>
'''

**Pull request merged**